File: skill-packs/java/spring-boot/spring-boot-enterprise-foundation/examples/good-example.java
// GOOD — constructor injection, layered entry, Actuator-ready service
@SpringBootApplication
public class OrderApplication {
    public static void main(String[] args) {
        SpringApplication.run(OrderApplication.class, args);
    }
}

@RestController
@RequestMapping("/api/v1/orders")
@RequiredArgsConstructor
class OrderController {
    private final OrderService orderService;

    @PostMapping
    ResponseEntity<OrderResponse> create(@Valid @RequestBody CreateOrderRequest request) {
        return ResponseEntity.status(HttpStatus.CREATED)
            .body(OrderResponse.from(orderService.create(request)));
    }
}

File: skill-packs/kotlin/spring/kotlin-spring-boot/examples/good-example.kt
// GOOD — Kotlin data class DTO, coroutine controller, null-safe service
@RestController
@RequestMapping("/api/v1/orders")
class OrderController(private val service: OrderService) {
    @PostMapping
    suspend fun create(@Valid @RequestBody request: CreateOrderRequest): ResponseEntity<OrderResponse> {
        val order = service.create(request)
        return ResponseEntity.status(HttpStatus.CREATED).body(order.toResponse())
    }
}
